### ISSUE TITLE
Audio randomly stopping

### ISSUE THREAD

**Issue by riples** [danielstern/ngAudio#134]
Not sure if anyone else has experienced this, but I am finding that my audio stream is constantly 'randomly' stopping after any given time.  I get no errors or alerts to notify me that something is wrong and my network connectivity remains connected.  Clicking play again intantly works fine.

My setup is simple as I've basically copied the setup from the documentation on the website.  I simply load a webstream URL.

I am using Chrome to play my content.

**Comment by danielstern** [comment]
Can you share a link to your implementation?

**Comment by Looooong** [comment]
I'm experiencing the the same case. This is not random to me. Here is what I do:
+ After loading web, load the audio and begin auto `play()`
+ Click on arbitrary spot on the screen and audio is paused. Debug for audio HTML element shows: `audio.paused = true`

**Comment by thewildandy** [comment]
I've had the same thing -- load an arbitrary clip, begin playing, the next mouse click causes the audio to pause. Subsequent play / pause works fine.

**Comment by thewildandy** [comment]
The problem is caused by the "unlock" feature. It affects usage of the ngAudio service from a controller. As far as I'm aware this doesn't affect the directive, but I may be wrong and haven't tested it, so you'll need to do your own tests if you're having trouble. 

#### TL;DR Solution:
Turn off the "unlock" setting.
*Be aware that additional work will be required for smooth operation on mobile devices.*

```
var myAudio = ngAudio.load('/path/to/file.mp3');
myAudio.unlock = false;
myAudio.play();
```

#### More Detail
The issue stems from this block:
```
if (ngAudioGlobals.unlock) {
    window.addEventListener("click",function twiddle(){
        audio.play();
        audio.pause();
        window.removeEventListener("click",twiddle);
    });
}
```

If I'm understanding your intentions correctly the purpose of this is to get around the issue of mobile browsers not playing the audio on the first request; you call `play()` then immediately call `pause()`, then subsequent calls to `play()` should work as intended. 

One problem with this is that, in my experience at least, this behavour differs greatly by device, particularly between Android and iOS platforms. In addition, if the audio is already playing before the mouseclick event fires the audio becomes paused and does not resume, which breaks the experience in browsers. Not good!

You might want to think about reworking this to take into account the user agent and behave appropriately depending on the browser. If you don't want to test specifically which devices _need_ the unlock feature, a quick hotfix to get browsers working and mobile behaviour unchanged would be to simply skip this block if the device isn't mobile.

The solution in my case, coming from the audiojs library and simply dropping ngAudio into an existing custom player as the file handler, is to turn off the "unlock" setting and use our own user-agent checks (which we've already written for most of our supported devices) to bust the loading issue on a per-device basis.

**Comment by Looooong** [comment]
@thewildandy Yes, I have just found that `unlock()` causes the audio to stop. So I puts a simple hack:
```js
// Try to click on arbitrary element
// Should avoid element with onclick callback
angular.element(document.body).click();
$scope.audio = ngAudio.load('/path/to/file.mp3').play();
```
